### PR TITLE
Rewrite workflow using reusable actions

### PR DESCRIPTION
**Pull request by archanaserver**
Updating the Ruby test workflow by using shared rubocop and foreman plugin test action:
https://github.com/theforeman/actions#rubocop
https://github.com/theforeman/actions#foreman-plugin-tests

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,93 +1,32 @@
...
-        run: |
-          bundle install --jobs=3 --retry=3
-          bundle exec rake db:create
-          bundle exec rake db:migrate
-          npm install
-          bundle exec rake webpack:compile
-      - name: Run plugin tests
-        run: |
-          bundle exec rake test:foreman_rescue
-          bundle exec rake test TEST="test/unit/foreman/access_permissions_test.rb"
```
Comment (archanaserver) [inline .github/workflows/ci.yml:89]: related discussion https://github.com/theforeman/actions/pull/16

**Comment by laugmanuel** [comment]
LGTM, however I'll wait until the discussion regarding `access_permissions_test` is solved.

**Comment by ekohl** [comment]
If you have input on the discussion, then that would be very welcome. I haven't seen a lot of input from plugin maintainers on what is desired.

**Comment by archanaserver** [comment]
I've temporarily drafted this until the issue at https://github.com/theforeman/actions/pull/16 is resolved

**Comment by laugmanuel** [comment]
This will be part of https://github.com/dm-drogeriemarkt/foreman_rescue/pull/27

**Pull request closed without merging**